config/config.go

```diff
@@ -298,3 +345,142 @@ func (rc *ReporterConfig) NewReporter(
 func (rc *ReporterConfig) newTransport() (jaeger.Transport, error) {
 	return jaeger.NewUDPTransport(rc.LocalAgentHostPort, 0)
 }
+
+// NewTracer returns a new tracer based on the current configuration, using the given options
+func (c *Configuration) NewTracer(options ...Option) (opentracing.Tracer, io.Closer, error) {
+	if c.ServiceName == "" {
+		return nil, nil, errors.New("no service name provided")
+	}
+
+	return c.New(c.ServiceName, options...)
+}
+
+// FromEnv uses environment variables to set the tracer's Configuration
+func FromEnv() *Configuration {
+	c := &Configuration{}
+
+	if e := os.Getenv(envServiceName); e != "" {
+		// We are ignoring any errors, so that we have a single return value
+		// This pattern repeats itself across all the "FromEnv" functions.
+		// TODO: should we at least log when something bad happens, instead of just ignoring?
+		c.ServiceName = e
+	}
+
+	if e := os.Getenv(envRPCMetrics); e != "" {
+		if value, err := strconv.ParseBool(e); err == nil {
+			c.RPCMetrics = value
+		}
+	}
+
+	if e := os.Getenv(envDisabled); e != "" {
+		if value, err := strconv.ParseBool(e); err == nil {
+			c.Disabled = value
+		}
+	}
+
+	if e := os.Getenv(envTags); e != "" {
+		c.Tags = parseTags(e)
+	}
+
+	c.Sampler = SamplerConfigFromEnv()
+	c.Reporter = ReporterConfigFromEnv()
+	return c
+}
+
+// SamplerConfigFromEnv creates a new SamplerConfig based on the environment variables
+func SamplerConfigFromEnv() *SamplerConfig {
```

Comment:
nit: how about we keep these private for now? Can always expose later if someone actually needs that